apps/src/lab2/levelEditors/aichatSettings/ModelSelectionFields.tsx

```diff
@@ -0,0 +1,105 @@
+import React, {useContext, useState, useCallback} from 'react';
+
+import {BodyFourText} from '@cdo/apps/componentLibrary/typography';
+import Checkbox from '@cdo/apps/componentLibrary/checkbox/Checkbox';
+import SimpleDropdown from '@cdo/apps/componentLibrary/dropdown/simpleDropdown';
+
+import {modelDescriptions} from '@cdo/apps/aichat/constants';
+import moduleStyles from './edit-aichat-settings.module.scss';
+import {UpdateContext} from './UpdateContext';
+import FieldSection from './FieldSection';
+import {Visibility} from '@cdo/apps/aichat/types';
+
+const modelDropdownItems = modelDescriptions.map(model => {
+  return {value: model.id, text: model.name};
+});
+
+const ModelSelectionFields: React.FunctionComponent = () => {
+  const {setModelSelectionValues, aichatSettings} = useContext(UpdateContext);
+  const shouldDisableAdditionalModelSelection =
+    aichatSettings.visibilities.selectedModelId !== Visibility.EDITABLE;
+  const [additionalAvailableModelIds, setAdditionalAvailableModelIds] =
+    useState<Set<string>>(
+      new Set(
```

Comment:
Would it be easier to just work with an array here? Realistically i don't think we'd have duplicated IDs right? And if we did I think it would cause a whole lot of problems elsewhere too